code-challenges.test.js

```diff
@@ -22,24 +30,96 @@ const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
 // b) Create the function that makes the test pass.
+// Input: takes in a string
+// Output: retunrs a string that has been encrypted.
+// Psuedo Code: create a function that turn my string into a array so i can change aech value per index using .split to break it into an array. use an if statement to iterate thru the array and change the value based on {  "a" to 4, "e" to 3, "i" to 1, and "o" to 0.  } 
+
+
+
+const theCoder = (string)=> {
+    newArray = string.split('')
+    
+    
+    const modifiedArray = newArray.map((char) => {
+        if (char == "a") {
+          return "4";
+        } else if (char == "E") {
+          return "3";
+        } else if (char == "i") {
+          return "1";
+        } else if (char == "o") {
+          return "0";
+        } else if (char == "e") {
+            return "3";
+        } else {
+          return char;
+        }
+      });
+    
+      return modifiedArray.join("");
+    };
+    
+    console.log(theCoder(secretCodeWord1)); // "L4ck4d41s1c4l"
+    console.log(theCoder(secretCodeWord2))
+    console.log(theCoder(secretCodeWord3))
+    
+    
+    
+    
+
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+describe("partLetter", () => {
+    it("returns an array of all the words containing that particular letter.", () => {
+      expect(partLetter(fruitArray,letterA)).toEqual(["Mango", "Apricot", "Peach"])
+      expect(partLetter(fruitArray, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+  })
+
+
+
 
-const letterA = "a"
 // Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
+
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
 
+// Input: an array
+// Output: returns the indexes that had the same char
+// Psuedo Code: makea  function that will check to see if the array value has a specific letter with in it and return that value back . 
+// i can use the .filter method to filter out any index that doesnt have it.
```

Comment:
This would be a great place to talk about using includes and the toLowerCase method to account for casing of a letter.